packages/block-library/src/navigation/edit/use-convert-classic-menu-to-block-menu.js

```diff
@@ -88,47 +87,42 @@ function useConvertClassicToBlockMenu( clientId ) {
 		return navigationMenu;
 	}
 
-	const convert = useCallback(
-		( menuId, menuName ) => {
-			if ( ! menuId || ! menuName ) {
-				setError( 'Unable to convert menu. Missing menu details.' );
+	const convert = useCallback( async ( menuId, menuName ) => {
+		if ( ! menuId || ! menuName ) {
+			setError( 'Unable to convert menu. Missing menu details.' );
+			setStatus( CLASSIC_MENU_CONVERSION_ERROR );
+			return;
+		}
+
+		setStatus( CLASSIC_MENU_CONVERSION_PENDING );
+		setError( null );
+
+		return await convertClassicMenuToBlockMenu( menuId, menuName )
```

Comment:
Now we return the value and allow the caller to handle it rather than use `setValue`. This feels simpler to me.

I think the same could be done for errors and status too, but I've tried to keep refactoring to a minimum.

Reply:
It doesn't look like I had a lot of input into those PRs. My comment is referenced on that earlier PR, so maybe it's that - https://github.com/WordPress/gutenberg/pull/38824#pullrequestreview-883918504.

My suggestion there was to make it an async function called `convertClassicToBlockMenu` instead of a hook, and it looks like it went the other way. That's absolutely fine, it's just a difference of opinion.

I can go into more detail about why I feel that way. I think the problem with relying on a hook is that creating a menu is a one time operation, but a hook generally returns values that are long lived (over multiple render).

So I might create a menu once by calling the `convert` function:
```js
const [ value, convert ] = useConvertClassicToBlockMenu( // ... );

// ...

onSelectClassicMenu={ ( classicMenu) => {
    convert( classicMenu );
} }
```

But because of the way hooks work, `value` continues to be set to the value of the menu for a long time after it is actually useful. Not only that, but it has to be handled in a `useEffect` hook separately from the code that called it.

If more work is done in the event handler, then that `value` can be garbage collected.